Subject: netiucv: reinsert dev_alloc_name for device naming
From: Stefan Raspl <raspl@linux.vnet.ibm.com>
Patch-mainline: v3.2-rc3
Git-commit: 1d503563f7601a249d015d3fad40b08e8d6a394b
References: bnc#792656,LTC#87086

Symptom:      When trying to setup multiple netiucv devices on SLES11-SP2,
              only the first device is really set up. For the remaining
              ones, errors like the following appear:
                 kernel: sysfs: cannot create duplicate filename
                 '/devices/iucv/netiucv%d'
Problem:      A call to re-allocate the name for the additional IUCV devices
              was inadvertently removed.
Solution:     Reinsert the call.
Reproduction: Setup multiple netiucv devices.

Signed-off-by: Stefan Raspl <raspl@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
--- a/drivers/s390/net/netiucv.c
+++ b/drivers/s390/net/netiucv.c
@@ -1994,6 +1994,8 @@ static struct net_device *netiucv_init_n
 			   netiucv_setup_netdevice);
 	if (!dev)
 		return NULL;
+	if (dev_alloc_name(dev, dev->name) < 0)
+		goto out_netdev;
 
 	privptr = netdev_priv(dev);
 	privptr->fsm = init_fsm("netiucvdev", dev_state_names,
